client/allocrunner/taskrunner/envoy_version_hook.go

```diff
@@ -95,7 +108,18 @@ func (h *envoyVersionHook) Prestart(_ context.Context, request *ifs.TaskPrestart
 	// Set the resulting image.
 	h.logger.Trace("setting task envoy image", "image", image)
 	request.Task.Config["image"] = image
-	response.Done = true
+	h.done = true
+	return nil
+}
+
+// PreKilling implements the ifs.TaskPreKillHook interface.
+//
+// When a sidecar task is restarting, it is possible a previous non-destructive
```

Comment:
"is restarting" seems a little ambiguous here because I think we'd only get `PreKilling` if the task has been intentionally restarted? (Or restarted because a separate main task failed, too, IIRC.) If the task has exited on its own and is getting restarted because of that, we're keeping `h.done = true` in place because we don't want to do the expensive request to Consul again.

That being said, is it possible a non-destructive update could change the image modification for that case as well? Or in other words, would it make more sense not to track `h.done` at all and take the expense of re-running the request to Consul every time?

Reply:
> Or in other words, would it make more sense not to track h.done at all and take the expense of re-running the request to Consul every time?

I'd be +1 on this.

I think we could modify the `skip()` func to mark `response.Done` for tasks we know will never need this hook to be done (eg non-docker non-sidecar tasks), but otherwise let's just always run it? :shrug:

Reply:
I have ripped out the internal tracking, which keeps it all a lot simpler.